frontend/lib/js/category-trees/globalTreeStoreHelper.js

```diff
@@ -120,3 +120,95 @@ export const hasConceptChildren = node => {
 
   return concept && concept.children && concept.children.length > 0;
 };
+
+/*
+  This is async because ... we might want to parallelize this very soon,
+  as there are up to 200k concepts that need to be searched.
+*/
+export const search = async (query: string) => {
+  const result = Object.keys(window.categoryTrees).reduce(
+    (all, key) => ({
+      ...all,
+      ...findConcepts(key, key, window.categoryTrees[key][key], query, {})
+    }),
+    {}
+  );
+
+  return result;
+};
+
+const doesQueryMatchNode = (node, query) => {
+  const lowerQuery = query.toLowerCase();
+
+  return (
```

Comment:
Du erzeugst hier viele kurzlebige objekte, wäre es nicht sinnvoller die irgendwie zu cachen oä? Oder zumindest aufbereitet vorzulegen?

Reply:
Welche Objekte meinst du? 

Wenn ich jetzt raten müsste: Theoretisch könnte man die "geLowerCased" `label` / `description` und die "gejointen" `additionalInfos` vorher berechnen / cachen. Weil die Suche aber sowieso so schnell ist, ist das denke ich nicht notwendig.

Reply:
Um noch konkreter zu sein: Hier ein Benchmark, Javascript ist relativ zügig bei den Operationen. Selbst beim langsameren join + lowercase Beispiel schafft mein Macbook 2,6 mio Operationen / Sekunde. Wir können also selbst auf einem 10x langsameren Rechner > 200k Ergebnisse in unter einer Sekunde durchsuchen. Würde sagen, das passt erstmal.

<img width="961" alt="Bildschirmfoto 2019-04-16 um 11 05 34" src="https://user-images.githubusercontent.com/1403093/56196710-a5a48080-6037-11e9-976c-1fa4295ea92e.png">